src/components/input-text/input-text.tsx

```diff
@@ -0,0 +1,554 @@
+import { Scale, Status } from "../interfaces";
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Method,
+  Prop,
+  VNode,
+  Watch
+} from "@stencil/core";
+import { getElementDir, getElementProp, getSlotted, setRequestedIcon } from "../../utils/dom";
+
+import { CSS, SLOTS, TEXT } from "./resources";
+import { Position } from "../interfaces";
+import { LabelableComponent, connectLabel, disconnectLabel, getLabelText } from "../../utils/label";
+import {
+  connectForm,
+  disconnectForm,
+  FormComponent,
+  HiddenFormInputSlot,
+  submitForm
+} from "../../utils/form";
+import { CSS_UTILITY, TEXT as COMMON_TEXT } from "../../utils/resources";
+
+import { createObserver } from "../../utils/observers";
+import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
+
+type SetValueOrigin = "initial" | "connected" | "user" | "reset" | "direct";
+
+/**
+ * @slot action - A slot for positioning a button next to the component.
+ */
+@Component({
+  tag: "calcite-input-text",
+  styleUrl: "input-text.scss",
+  shadow: true
+})
+export class InputText implements LabelableComponent, FormComponent, InteractiveComponent {
```

Comment:
One more 'next step' comment, we should start thinking about extracting common behavior, rendering, etc so all input types can leverage.